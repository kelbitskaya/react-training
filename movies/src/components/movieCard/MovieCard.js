import React, { useState } from 'react';
import PropTypes from 'prop-types';
import MovieMenu from '../movieMenu/MovieMenu';
import MovieDeletePopup from '../movieDeletePopup/movieDeletePopup';
import MovieEditPopup from '../movieEditPopup/movieEditPopup';

export default function MovieCard(props) {
  const {
    title, genre, year, src, id, releaseDate, url, overview, runtime,
    handleClose,
  } = props;

  const [deleteMovie, deleteMovieOpen] = useState(false);
  const [editMovie, editMovieOpen] = useState(false);

  const deleteMoviePopup = () => {
    deleteMovieOpen(!deleteMovie);
  };
  const editMoviePopup = () => {
    editMovieOpen(!editMovie);
  };

  return (
    <div className="movie-card">
      <MovieMenu
        deleteMovie={() => deleteMoviePopup()}
        editMovie={() => editMoviePopup()}
      />
      <MovieEditPopup
        isOpen={editMovie}
        handleClose={handleClose}
        title={title}
        genre={genre}
        releaseDate={releaseDate}
        src={src}
        url={url}
        overview={overview}
        runtime={runtime}
        id={id}
      />
      <MovieDeletePopup
        isOpen={deleteMovie}
        handleClose={handleClose}
      />
      <img src={src} alt={title} className="movie-card__img" />
      <div className="movie-card__info">
        <div className="movie-card__title-wrap">
          <h3 className="movie-card__title">{title}</h3>
          <p className="movie-card__descrp">{genre}</p>
        </div>
        <p className="movie-card__year">{year}</p>
      </div>
    </div>
  );
}

MovieCard.propTypes = {
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  runtime: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  genre: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleClose: PropTypes.func.isRequired,
};
