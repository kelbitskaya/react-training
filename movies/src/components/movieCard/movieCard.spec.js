import React from 'react';
import MovieCard from './MovieCard';
import {Provider} from "react-redux";
import { createMockStore } from 'redux-test-utils';
import rerender from "react-test-renderer";
import routeData from "react-router";

let store = createMockStore('');
const resetStore = () => store = createMockStore('');

const mockLocation = {
  pathname: '/welcome',
  hash: '',
  search: '',
  state: ''
};

const props = {
  title: '',
  genre: [],
  year: 123,
  src: '',
  id: '',
  releaseDate: "123",
  url: '',
  overview: '',
  runtime: '123',
  selectMovie: jest.fn(),
  description: '',
  rating: 123,
  getMovie: jest.fn(),
  isSubmitting: jest.fn(),

};

describe("should render MovieCard component", ()=>{
  beforeEach(() => {
    resetStore();
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
  });
  it("should contain movie-card wrapper", ()=> {
    const component = rerender.create(
      <Provider store={store}>
        <MovieCard {...props} />
      </Provider>
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


