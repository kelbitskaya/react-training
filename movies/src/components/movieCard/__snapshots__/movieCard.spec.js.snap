// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieCard component should contain movie-card wrapper 1`] = `
<div
  className="movie-card"
>
  <div
    className="menu__item"
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="circle"
    />
    <div
      className="circle"
    />
    <div
      className="circle"
    />
  </div>
  <img
    alt=""
    className="movie-card__img"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    src=""
  />
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__title-wrap"
    >
      <h3
        className="movie-card__title"
      >
        
      </h3>
      <p
        className="movie-card__rating"
      >
        123
      </p>
      <p
        className="movie-card__descrp"
      />
    </div>
    <div
      className="movie-card__time"
    >
      <p
        className="movie-card__year"
      >
        123
      </p>
      <p
        className="movie-card__runtime"
      >
        123
         min
      </p>
    </div>
    <div
      className="movie-card__full-descrp"
    >
      
    </div>
  </div>
</div>
`;
